kbs/src/resource/plugin/mod.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2024 by IBM Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(feature = "nebula-plugin")]
+mod nebula;
+
+use anyhow::{anyhow, bail, Context, Result};
+use serde::Deserialize;
+use std::fs;
+use std::path::Path;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+#[cfg(feature = "nebula-plugin")]
+use crate::resource::plugin::nebula::NebulaPluginConfig;
+
+trait PluginBuild {
+    fn get_plugin_name(&self) -> &str;
+    fn create_plugin(&self, work_dir: &str) -> Result<Arc<RwLock<dyn Plugin + Send + Sync>>>;
+}
+
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct PluginManagerConfig {
+    work_dir: String,
+    enabled_plugins: Vec<String>,
+}
+
+impl PluginManagerConfig {
```

Comment:
Are we sure the `PluginManager/PluginManagerConfig/PluginBuilder` abstractions are required? I am concerned they introduce accidental complexity without a use case. Afaict we don't really need to "manage" plugins, they are activated and configured at launch, and do not change at runtime.

IMO a plugin could be an implementation of this trait:

```rust
trait Plugin {
    async fn get_resource(&self, resource: &str, query_string: &str) -> Result<Vec<u8>>;
}
```

when parsing `KbsConfig` a Plugin can be initialized and registered in a lookup map.

```rust
// we don't want to have dynamic strings for plugin lookup, they are known at compile-time
const NEBULA: &str = "nebula";
...
// this needs to be in the app's state for lookup in the route handlers
let mut enabled_plugins: BTreeMap<&'static str, Box<dyn Plugin + Send + Sync>>;
...
let nebula_plugin = nebula::new(&kbs_config)?;
enabled_plugins.insert(NEBULA, Box::new(nebula_plugin));
```